server/db/models/cart.js

```diff
@@ -1,8 +1,17 @@
 var mongoose = require('mongoose');
 
+var lineItemSchema = new mongoose.Schema({
+	film: {type:mongoose.Schema.Types.ObjectId, ref: 'Film'},
+	quantity: {type: Number, default: 1, required: true}
+})
+
 var cartSchema = new mongoose.Schema({
-	films: [{type:mongoose.Schema.Types.ObjectId, ref: 'Film'}],
-	sessionId: {type: String, required: true}
+	lineItems: [lineItemSchema],
+	sessionId: {type: String, required: true},
+	promoApplied: {type: Boolean, required: true}, 
+	promoDiscount: {type: Number, required: true, default: 0},
+	closed: {type: Boolean, required: true, default: false}
```

Comment:
This is cool field - we "close" the Cart after an order has been processed so that it is no longer accessible - but we will still keep it in the DB just in case.